src/extension/src/ProcessHandler.py

```diff
@@ -55,13 +57,21 @@ def get_env_settings(self, ext_env_handler):
         return env_settings
 
     def start_daemon(self, seq_no, config_settings, ext_env_handler):
-        """ Launches the core code in a separate independent process with required arguements and exits the current process immediately """
+        """ Launches the core code in a separate independent process with required arguments and exits the current process immediately """
         exec_path = os.path.join(os.getcwd(), Constants.CORE_CODE_FILE_NAME)
         public_config_settings = base64.b64encode(json.dumps(self.get_public_config_settings(config_settings)).encode("utf-8")).decode("utf-8")
         env_settings = base64.b64encode(json.dumps(self.get_env_settings(ext_env_handler)).encode("utf-8")).decode("utf-8")
 
         args = " -sequenceNumber {0} -environmentSettings \'{1}\' -configSettings \'{2}\'".format(str(seq_no), env_settings, public_config_settings)
-        command = ["python " + exec_path + " " + args]
+
+        # Verify the python version available on the machine to use
+        self.logger.log("Python version: " + " ".join(sys.version.splitlines()))
+        if sys.version_info.major == 3:
+            python_cmd = "python3"
+        else:
+            python_cmd = "python"
```

Comment:
As discussed, please mirror the logic in shim (probe-based discovery). Real world behavior is not as clear cut as this code as it depends on what came with the distribution and/or what was installed after.

Also add additional suitable logging for the behavior seen where core does not seem to run. A good way to do that maybe to capture data from stdout and stderr for some brief time and log it into the handler log if any data is seen there.